addons/l10n_id/models/qris_transaction.py

```diff
@@ -0,0 +1,97 @@
+        return {
+            'amount': self.amount,
+            'payment_type': 'inbound',
+            'currency_id': move.currency_id.id,
+            'partner_id': move.partner_id.commercial_partner_id.id,
+            'partner_type': 'customer',
+            'journal_id': self.partner_bank_id.journal_id.id,
+            'ref': move.payment_reference,
+            **extra_create_values
+        }
+
+    def _cron_fetch_payment_status(self):
+        """ Get all qris transaction that is within 30 minutes and fetch each status"""
+        prev_30_minutes = fields.Datetime.now() - datetime.timedelta(minutes=30)
+        transactions = self.search([('paid', '=', False), ('trx_date', '>=', prev_30_minutes), ('model', '=', 'account.move')])
+        paid_transactions = transactions.fetch_payment_status()
+        paid_transactions.register_payment()
+        self._clean_transactions()
+
+    def fetch_payment_status(self):
+        """ Fetch the status of each transaction via API
+        """
+        paid_transactions = self.env['l10n_id.qris.transaction']
+        for transaction in self:
+            params = {
+                'do': 'checkStatus',
+                'apikey': transaction.partner_bank_id.l10n_id_qris_api_key,
+                'mID': transaction.partner_bank_id.l10n_id_qris_mid,
+                'invid': transaction.qris_invoiceid,
+                'trxvalue': int(transaction.amount),
+                'trxdate': transaction.trx_date.strftime('%Y-%m-%d')
+            }
+            response = self.partner_bank_id.make_qris_request(QRIS_FETCH_STATUS_URL, params)
+
+            # upate payment status of status repsonse is 'paid'
+            _logger.debug("Found response: %s", pprint.pformat(response))
+            payment_status = response['data'].get('qris_status', 'unpaid')
+
+            if payment_status == "paid":
+                transaction.paid = True
+                paid_transactions |= transaction
+            elif payment_status == "unpaid":
+                transaction.paid = False
+
+        return paid_transactions
+
+    def register_payment(self):
+        """ Registering the payments for invoice whose transaction is paid"""
+        for transaction in self:
+            if not transaction.paid:
+                continue
+
+            invoice = self.env[transaction.model].browse(transaction.res_id)
+
+            payment_vals = self._create_payment_vals(invoice)
+            payment = self.env['account.payment'].create(payment_vals)
+            payment.action_post()
+
+            if invoice.state == 'draft':
+                invoice.action_post()
+
+            (payment.line_ids + invoice.line_ids).filtered(
+                lambda line: line.account_id == payment.destination_account_id and not line.reconciled
+            ).reconcile()
+
+    def _clean_transactions(self):
+        """ Remove all unpaid transactions"""
+        unpaid_trx = self.search([('paid', '=', False)])
```

Comment:
This will delete all unpaid transactions even those currently processing
To me this method is useless and you can just get `transactions - paid_transactions` and unlink that

Reply:
with `transactions - paid_transactions` => these are those transactions that are generated within 30 minutes of the CRON call and still not paid yet. 
Wouldn't this mean that we're deleting transactions that is potentially being paid by customer currently?
I think then we should delete those transactions that are generated past 30 minutes already?

Reply:
yes right